src/tests/logic/calculate.spec.js

```diff
@@ -0,0 +1,10 @@
+import calculate from '../../logic/calculate';
+
+describe('tests for calculate function', () => {
+  test('show buttonName when it is clicked', () => {
+    const data = { total: 5, next: '', operation: '/' };
+    const newData = calculate(data, '/');
+    expect(newData.operation).toEqual('/');
+    expect(newData.next).toEqual('');
```

Comment:
- Add test for `AC` button. Check whether total = '', next = '' and operation = null when this button is clicked. It's always a good idea to fully test all functions in your code.
- Please add test for `+/-`, `=` buttons, and any other functionalities in the `calculate.js` module.